src/main/java/org/opensearch/flowframework/workflow/ReIndexStep.java

```diff
@@ -0,0 +1,175 @@
+
+        Set<String> optionalKeys = Set.of(REFRESH, REQUESTS_PER_SECOND, REQUIRE_ALIAS, SLICES, MAX_DOCS);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String sourceIndices = (String) inputs.get(SOURCE_INDEX);
+            String destinationIndex = (String) inputs.get(DESTINATION_INDEX);
+            Boolean refresh = inputs.containsKey(REFRESH) ? Booleans.parseBoolean(inputs.get(REFRESH).toString()) : null;
+            Integer requestsPerSecond = (Integer) inputs.get(REQUESTS_PER_SECOND);
+            Boolean requireAlias = inputs.containsKey(REQUIRE_ALIAS) ? Booleans.parseBoolean(inputs.get(REQUIRE_ALIAS).toString()) : null;
+            Integer slices = (Integer) inputs.get(SLICES);
+            Integer maxDocs = (Integer) inputs.get(MAX_DOCS);
+
+            ReindexRequest reindexRequest = new ReindexRequest();
+            reindexRequest.setSourceIndices(sourceIndices);
+            reindexRequest.setDestIndex(destinationIndex);
+            if (refresh != null) {
+                reindexRequest.setRefresh(refresh);
+            }
+            if (requestsPerSecond != null) {
+                reindexRequest.setRequestsPerSecond(requestsPerSecond);
+            }
+            if (requireAlias != null) {
+                reindexRequest.setRequireAlias(requireAlias);
+            }
+            if (maxDocs != null) {
+                reindexRequest.setMaxDocs(maxDocs);
+            }
+            if (slices != null) {
+                reindexRequest.setSlices(slices);
+            }
+
+            ActionListener<BulkByScrollResponse> actionListener = new ActionListener<>() {
+
+                @Override
+                public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+                    logger.info("Reindex from source: {} to destination {}", sourceIndices, destinationIndex);
+                    try {
+                        if (bulkByScrollResponse.getBulkFailures().isEmpty() && bulkByScrollResponse.getSearchFailures().isEmpty()) {
+                            flowFrameworkIndicesHandler.updateResourceInStateIndex(
+                                currentNodeInputs.getWorkflowId(),
+                                currentNodeId,
+                                getName(),
+                                destinationIndex,
+                                ActionListener.wrap(response -> {
+                                    logger.info("successfully updated resource created in state index: {}", response.getIndex());
+
+                                    reIndexFuture.onResponse(
+                                        new WorkflowData(
+                                            Map.of(RE_INDEX_FIELD, Map.of(sourceIndices, destinationIndex)),
```

Comment:
Why do we need to return the source and destination indices as workflow data?

Reply:
I expect we do (probably) need the destination index here, but not sure about source. Don't think it hurts to include them both, though.